src/elements/Label/Label.js

```diff
@@ -0,0 +1,168 @@
+import cx from 'classnames'
+import React, { createElement, PropTypes } from 'react'
+import createFragment from 'react-addons-create-fragment'
+
+import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import { someChildType } from '../../utils/childrenUtils'
+import {
+  iconPropRenderer,
+  imagePropRenderer,
+  useKeyOnly,
+  useValueAndKey,
+  useKeyOrValueAndKey,
+} from '../../utils/propUtils'
+import Icon from '../Icon/Icon'
+import Image from '../Image/Image'
+
+/**
+ * A label displays content classification
+ */
+const Label = (props) => {
+  const {
+    attached, children, color, corner, className, circular, detail, detailLink, floating, horizontal, icon,
+    image, link, onClick, onClickDetail, onClickRemove, pointing, removable, ribbon, size, tag, text,
+    ...rest,
+  } = props
+
+  const handleClick = e => onClick && onClick(e, props)
+  const handleClickRemove = e => onClickRemove && onClickRemove(e, props)
+  const handleClickDetail = e => onClickDetail && onClickDetail(e, props)
+
+  const _component = image || link || onClick ? 'a' : 'div'
+
+  const classes = cx('sd-label ui',
+    size,
+    color,
+    useKeyOnly(floating, 'floating'),
+    useKeyOnly(horizontal, 'horizontal'),
+    useKeyOnly(pointing, 'pointing'),
+    useKeyOnly(tag, 'tag'),
+    useValueAndKey(attached, 'attached'),
+    useValueAndKey(corner, 'corner'),
+    useKeyOrValueAndKey(pointing, 'pointing'),
+    useKeyOrValueAndKey(ribbon, 'ribbon'),
+    circular && (children && 'circular' || 'empty circular'),
+    // TODO how to handle image child with no image class? there are two image style labels.
+    (image || someChildType(children, Image) || someChildType(children, 'img')) && 'image',
+    'label',
+    className
+  )
+
+  const _props = {
+    className: classes,
+    onClick: handleClick,
+    ...rest,
+  }
+
+  const _detail = detail || detailLink
+  const detailComponent = (detailLink || onClickDetail) && 'a' || detail && 'div'
+
+  const _children = createFragment({
+    icon: iconPropRenderer(icon),
+    image: imagePropRenderer(image),
+    text,
+    children,
+    detail: _detail && createElement(detailComponent, { className: 'detail', onClick: handleClickDetail }, _detail),
+    remove: (removable || onClickRemove) && <Icon className='delete' onClick={handleClickRemove} />,
+  })
+
+  // Do not destructure createElement import
+  // react-docgen only recognizes a stateless component when React.createElement is returned
+  return React.createElement(_component, _props, _children)
+}
+
+Label._meta = {
+  library: META.library.semanticUI,
+  name: 'Label',
+  type: META.type.element,
+  props: {
+    attached: ['top', 'bottom', 'top right', 'top left', 'bottom left', 'bottom right'],
+    size: sui.sizes,
+    color: sui.colors,
+    pointing: ['bottom', 'left', 'right'],
+    corner: ['left', 'right'],
+    right: ['left', 'right'],
+    ribbon: ['right'],
+  },
```

Comment:
Enum props are documented in our `_meta` tag.  This allows us to write shared tests against them.  Also will provide better doc site features, like generating forms that control and demonstrate component functionality.